client/client-lib/src/lib.rs

```diff
@@ -430,23 +431,51 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
 }
 
 impl Client<UserClientConfig> {
-    pub async fn fetch_gateway(&self) -> Result<LightningGateway> {
-        // fetch gateway from db
-        if let Some(gateway) = self
-            .context
+    pub async fn fetch_registered_gateways(&self) -> Result<Vec<LightningGateway>> {
+        let gateways = self.context.api.fetch_gateways().await?;
+        let current_active = self.fetch_active_gateway().await;
+
+        if current_active.is_some() {
+            // We have an active gateway in this client. We should update the list of registered gateways with that information.
+            Ok(gateways
+                .into_iter()
```

Comment:
optimization: `into_iter` moves values and for what we're trying to do here we can use `iter_mut` (which yields `&mut T`)
I'll comment the full gist